### PR TITLE
draft for buildbot-worker

### PR DESCRIPTION
**Pull request by Akira25**
Role for handling buildbot-workers.

### CODE REVIEW

**Comment by rtznprmpftl** [comment]
please rename `buildbot-worker` to `buildbot_worker`

https://galaxy.ansible.com/docs/contributing/creating_role.html#role-names

Role names are limited to lowercase word characters (i.e., a-z, 0-9) and ‘_’. No special characters are allowed, including ‘.’, ‘-‘, and space.

File: roles/buildbot-worker/handlers/main.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+- name: buildbot-worker | restart
+  systemd:
+    daemon_reload: yes
+    name: buildbot-worker
+    enabled: true
+    state: restarted
+
+- name: buildbot-worker | initialise
+  command: buildbot-worker create-worker --umask=0o22 {{ bbworker_dir }} {{ bbmaster_fqdn }}:{{ bbmaster_port }} {{ bbworker_name }} {{ bbworker_pwd }}
```
Comment (rtznprmpftl) [inline roles/buildbot-worker/handlers/main.yml:9]: the 0o22 seems like a typo

File: roles/buildbot-worker/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
+---
+
+- name: install dependencies for falter build system and openwrt
```
Comment (rtznprmpftl) [inline roles/buildbot-worker/tasks/main.yml:3]: do we need both python2 and 3?

File: roles/buildbot-worker/handlers/main.yml (hunk 1 above)
Comment (Akira25) [reply roles/buildbot-worker/handlers/main.yml:9]: Looks indeed a bit strange, but it is intended like that.

https://docs.buildbot.net/latest/manual/installation/worker.html#worker-options

File: roles/buildbot-worker/tasks/main.yml (hunk 2 above)
Comment (Akira25) [reply roles/buildbot-worker/tasks/main.yml:3]: I think so, as they are both defined as dependencies for the build-system:
https://openwrt.org/docs/guide-developer/toolchain/install-buildsystem#debianubuntu

Imagebuilder needs to have them installed too.

**Comment by Akira25** [comment]
> please rename `buildbot-worker` to `buildbot_worker`
> 
> https://galaxy.ansible.com/docs/contributing/creating_role.html#role-names
> 
> Role names are limited to lowercase word characters (i.e., a-z, 0-9) and ‘_’. No special characters are allowed, including ‘.’, ‘-‘, and space.

renamed in new commit-version.

**Review by rtznprmpftl — APPROVED** [review]

File: inventory/hosts [hunk 3]
```diff
@@ -3,3 +3,8 @@ buildbot.berlin.freifunk.net
 
 [configserver]
 config-dev.berlin.freifunk.net
+
+[buildbotworker]
```
[the remark this answers is not in the document]
Comment (Akira25) [reply inventory/hosts:7]: > buildbot-worker-1 in the hosts file and in the host_var you can add a `ansible_host: 85.215.202.123`

On my machine it didn't work to have the `ansible_host` in the encrypted host-vars. Instead I've added them in the inventory now, which should have the same effect.

**Review by rtznprmpftl — APPROVED** [review]

**Comment by Akira25** [comment]
Merged due approval.

**Pull request merged**